### PR TITLE
Ensure electron frontends cannot access files outside of installer path

### PR DESCRIPTION
**Pull request by adamhock**
Electron users cannot access files outside of installer path

### CODE REVIEW

**Review by DanEastBentley — CHANGES_REQUESTED** [review]
Run "rush clean" before "rush build" and "rush extract-api" to prevent these false changes in the *.md files.

File: core/electron-manager/src/backend/ElectronHost.ts [hunk 1]
```diff
@@ -114,7 +114,12 @@ export class ElectronHost {
     let assetPath = requestedUrl.substr(this._electronFrontend.length);
     if (assetPath.length === 0)
       assetPath = "index.html";
-    assetPath = assetPath.replace(/(#|\?).*$/, "");
+    let regexp = ["..", "#", "?", "*", "$"];
```
Comment (wgoehrig) [inline core/electron-manager/src/backend/ElectronHost.ts:117]: I think there's probably a simpler way to do this...

Note that below, we use `fs.realpathSync` below to get a fully resolved _absolute_ path.  So we don't need to ban relative paths or other characters, we just need to check that the resolved path is still inside webResourcesPath, right?

File: core/electron-manager/src/backend/ElectronHost.ts [hunk 2]
```diff
@@ -128,6 +127,8 @@ export class ElectronHost {
       // eslint-disable-next-line no-console
       // console.warn(`WARNING: Frontend requested "${requestedUrl}", but ${assetPath} does not exist`);
     }
+    if (!assetPath.startsWith(this.webResourcesPath))
+      throw new Error("Access outside that path is prohibited");
```
Comment (wgoehrig) [inline core/electron-manager/src/backend/ElectronHost.ts:131]: "that path" is ambiguous - how does:

```suggestion
      throw new Error(`Access to files outside installation directory (${this.webResourcesPath}) is prohibited`);
```

sound?

**Review by DanEastBentley — APPROVED** [review]

File: core/electron-manager/src/backend/ElectronHost.ts [hunk 3]
```diff
@@ -114,7 +114,6 @@ export class ElectronHost {
     let assetPath = requestedUrl.substr(this._electronFrontend.length);
     if (assetPath.length === 0)
       assetPath = "index.html";
-    assetPath = assetPath.replace(/(#|\?).*$/, "");
 
     // NEEDS_WORK: Remove this after migration to DesktopAuthorizationClient
```
Comment (kabentley) [inline core/electron-manager/src/backend/ElectronHost.ts:118]: @ramanujam-raman do you know what this is? Can it be removed?

**Review by wgoehrig — APPROVED** [review]

**Pull request merged**